benchmark/single-source/DataBenchmarks.swift

```diff
@@ -0,0 +1,393 @@
+        0x69,0x6e,0x63,0x69, 0x64,0x69,0x64,0x75,  0x6e,0x74,0x20,0x75, 0x74,0x20,0x6c,0x61,
+        0x62,0x6f,0x72,0x65, 0x20,0x65,0x74,0x20,  0x64,0x6f,0x6c,0x6f, 0x72,0x65,0x20,0x6d,
+        0x61,0x67,0x6e,0x61, 0x20,0x61,0x6c,0x69,  0x71,0x75,0x61,0x2e, 0x20,0x55,0x74,0x20,
+        0x65,0x6e,0x69,0x6d, 0x20,0x61,0x64,0x20,  0x6d,0x69,0x6e,0x69, 0x6d,0x20,0x76,0x65,
+        0x6e,0x69,0x61,0x6d, 0x2c,0x0a,0x71,0x75,  0x69,0x73,0x20,0x6e, 0x6f,0x73,0x74,0x72,
+        0x75,0x64,0x20,0x65, 0x78,0x65,0x72,0x63,  0x69,0x74,0x61,0x74, 0x69,0x6f,0x6e,0x20,
+        0x75,0x6c,0x6c,0x61, 0x6d,0x63,0x6f,0x20,  0x6c,0x61,0x62,0x6f, 0x72,0x69,0x73,0x20,
+        0x6e,0x69,0x73,0x69, 0x20,0x75,0x74,0x20,  0x61,0x6c,0x69,0x71, 0x75,0x69,0x70,0x20,
+        0x65,0x78,0x20,0x65, 0x61,0x20,0x63,0x6f,  0x6d,0x6d,0x6f,0x64, 0x6f,0x0a,0x63,0x6f,
+        0x6e,0x73,0x65,0x71, 0x75,0x61,0x74,0x2e,  0x20,0x44,0x75,0x69, 0x73,0x20,0x61,0x75,
+        0x74,0x65,0x20,0x69, 0x72,0x75,0x72,0x65,  0x20,0x64,0x6f,0x6c, 0x6f,0x72,0x20,0x69,
+        0x6e,0x20,0x72,0x65, 0x70,0x72,0x65,0x68,  0x65,0x6e,0x64,0x65, 0x72,0x69,0x74,0x20,
+        0x69,0x6e,0x20,0x76, 0x6f,0x6c,0x75,0x70,  0x74,0x61,0x74,0x65, 0x20,0x76,0x65,0x6c,
+        0x69,0x74,0x20,0x65, 0x73,0x73,0x65,0x0a,  0x63,0x69,0x6c,0x6c, 0x75,0x6d,0x20,0x64,
+        0x6f,0x6c,0x6f,0x72, 0x65,0x20,0x65,0x75,  0x20,0x66,0x75,0x67, 0x69,0x61,0x74,0x20,
+        0x6e,0x75,0x6c,0x6c, 0x61,0x20,0x70,0x61,  0x72,0x69,0x61,0x74, 0x75,0x72,0x2e,0x20,
+        0x45,0x78,0x63,0x65, 0x70,0x74,0x65,0x75,  0x72,0x20,0x73,0x69, 0x6e,0x74,0x20,0x6f,
+        0x63,0x63,0x61,0x65, 0x63,0x61,0x74,0x20,  0x63,0x75,0x70,0x69, 0x64,0x61,0x74,0x61,
+        0x74,0x20,0x6e,0x6f, 0x6e,0x0a,0x70,0x72,  0x6f,0x69,0x64,0x65, 0x6e,0x74,0x2c,0x20,
+        0x73,0x75,0x6e,0x74, 0x20,0x69,0x6e,0x20,  0x63,0x75,0x6c,0x70, 0x61,0x20,0x71,0x75,
+        0x69,0x20,0x6f,0x66, 0x66,0x69,0x63,0x69,  0x61,0x20,0x64,0x65, 0x73,0x65,0x72,0x75,
+        0x6e,0x74,0x20,0x6d, 0x6f,0x6c,0x6c,0x69,  0x74,0x20,0x61,0x6e, 0x69,0x6d,0x20,0x69,
+        0x64,0x20,0x65,0x73, 0x74,0x20,0x6c,0x61,  0x62,0x6f,0x72,0x75, 0x6d,0x2e,0x0a,0x00]
+    return Data(bytes: bytes)
+}
+
+func sampleBridgedNSData() -> Data {
+    let count = 1033
+    var bytes = [UInt8](repeating: 0, count: count)
+    bytes.withUnsafeMutableBufferPointer {
+        arc4random_buf($0.baseAddress, $0.count)
+    }
+    let data = NSData(bytes: bytes, length: count)
+    return Data(referencing: data)
+}
+
+func sampleData(_ type: SampleKind) -> Data {
+    switch type {
+    case .small: return sampleData(size: 11)
+    case .medium: return sampleData(size: 1033)
+    case .large: return sampleData(size: 40980)
+    case .veryLarge: return sampleData(size: 1024 * 1024 * 1024 + 128)
+    case .string: return sampleString()
+    case .immutableBacking: return sampleBridgedNSData()
+    }
+    
+}
+
+func benchmark_Subscript(_ N: Int, _ data: Data, _ index: Data.Index) {
+    for _ in 1...N {
```

Comment:
should this be 10000*N?

Also: you use 10000*N for all benchmarks. But I assume the time for one iteration varies a lot, e.g. for benchmark_Count vs benchmark_copyBytes.
Did you check (by running it locally) that the reported time is within a reasonable range for all benchmarks? That means > 10, ideally > 100.

Reply:
One iteration gives a value of 0, so yah I had to scale it up.

Reply:
Here is what I get locally with the suggested changes:

```
#,TEST,SAMPLES,MIN(us),MAX(us),MEAN(us),SD(us),MEDIAN(us)
1,DataAccessBytes,15,1276,1397,1328,42,1334
2,DataAppendArray,15,6051,6350,6237,95,6268
3,DataAppendBytes,15,5929,6348,6082,110,6072
4,DataAppendDataLargeToLarge,15,59606,77069,65998,5253,65009
5,DataAppendDataLargeToMedium,15,34978,37283,35716,739,35404
6,DataAppendDataLargeToSmall,15,33598,40655,35087,1628,34808
7,DataAppendDataMediumToLarge,15,22576,34445,24782,2934,23565
8,DataAppendDataMediumToMedium,15,8311,8848,8517,151,8480
9,DataAppendDataMediumToSmall,15,7897,8755,8174,200,8149
10,DataAppendDataSmallToLarge,15,22324,33598,24158,3589,22807
11,DataAppendDataSmallToMedium,15,7990,8534,8230,165,8223
12,DataAppendDataSmallToSmall,15,7574,8004,7784,129,7757
13,DataAppendSequence,15,19224,20106,19506,282,19471
14,DataCopyBytes,15,3007,3229,3123,56,3122
15,DataCount,15,52,54,52,1,53
16,DataMutateBytes,15,4776,4950,4868,52,4862
17,DataReplaceLarge,15,40273,43275,41635,1009,41217
18,DataReplaceLargeBuffer,15,56077,59662,57944,878,57766
19,DataReplaceMedium,15,13324,13753,13549,123,13537
20,DataReplaceMediumBuffer,15,13211,13734,13461,130,13459
21,DataReplaceSmall,15,10941,11486,11143,132,11146
22,DataReplaceSmallBuffer,15,10484,11222,10690,181,10711
23,DataReset,15,3397,3787,3579,102,3585
24,DataSetCount,15,919,1027,975,33,974
25,DataSubscript,15,284,297,288,4,289
```

Reply:
Oh that's 15 samples.